pkg/collector/corechecks/system/disk_nix.go

```diff
@@ -121,23 +129,22 @@ func (c *DiskCheck) sendPartitionMetrics(sender aggregator.Sender, usage *disk.U
 	sender.Gauge(fmt.Sprintf(diskMetric, "total"), float64(usage.Total)/1024, "", tags)
 	sender.Gauge(fmt.Sprintf(diskMetric, "used"), float64(usage.Used)/1024, "", tags)
 	sender.Gauge(fmt.Sprintf(diskMetric, "free"), float64(usage.Free)/1024, "", tags)
-	// FIXME: 6.x, use percent, a lot more logical than in_use
+	// FIXME(8.x): use percent, a lot more logical than in_use
 	sender.Gauge(fmt.Sprintf(diskMetric, "in_use"), usage.UsedPercent/100, "", tags)
 
 	// Inodes metrics
 	sender.Gauge(fmt.Sprintf(inodeMetric, "total"), float64(usage.InodesTotal), "", tags)
 	sender.Gauge(fmt.Sprintf(inodeMetric, "used"), float64(usage.InodesUsed), "", tags)
 	sender.Gauge(fmt.Sprintf(inodeMetric, "free"), float64(usage.InodesFree), "", tags)
-	// FIXME: 6.x, use percent, a lot more logical than in_use
+	// FIXME(8.x): use percent, a lot more logical than in_use
 	sender.Gauge(fmt.Sprintf(inodeMetric, "in_use"), usage.InodesUsedPercent/100, "", tags)
 
 }
 
-func (c *DiskCheck) sendDiskMetrics(sender aggregator.Sender, ioCounter disk.IOCountersStat, tags []string) {
-
-	// /1000 as psutil returns the value in ms
-	// Rate computes a rate of change between to consecutive check run.
-	// For cumulated time values like read and write times this a ratio between 0 and 1, we want it as a percentage so we *100 in advance
+func (c *DiskCheck) sendDiskMetrics(sender aggregator.Sender, ioCounter, lastIoCounter disk.IOCountersStat, tags []string) {
+	sender.Count(fmt.Sprintf(diskMetric, "read_time"), float64(incrementWithOverflow(ioCounter.ReadTime, lastIoCounter.ReadTime)), "", tags)
+	sender.Count(fmt.Sprintf(diskMetric, "write_time"), float64(incrementWithOverflow(ioCounter.WriteTime, lastIoCounter.WriteTime)), "", tags)
```

Comment:
Could we maybe use `sender.MonotonicCount` here? It handles `float64` so I suppose overflow is less of a concern. Not sure if its doable/advisable though.

Reply:
It's the underlying counter that can overflow, not the metric. When that happens, this code handles it better than MonotonicCount because `incrementWithOverflow` is able to report the actual difference between the current and previous value (MonotonicCount just ignores a value if it's lower than the previous one). On the other hand, it would greatly simplify the code and tests... so I'm undecided. 😂🤔 

On the python implementation where we don't have `incrementWithOverflow`, however, I think we should use MonotonicCounter 👍

Reply:
Oh, I see. I suppose it depends on how common the overflow situation is then.